packages/app/src/systems/Transaction/services/transaction.tsx

```diff
@@ -214,13 +214,24 @@ export class TxService {
       amount: isEth(coin) ? bn(coin.amount).add(gasFee) : bn(coin.amount),
     }));
 
+    if (!coins.find(isEth)) {
+      coins.push({
+        assetId: NativeAssetId,
+        amount: gasFee,
+      });
+    }
+
```

Comment:
as I'm understanding this code is made to add UTXO input to pay gas fee, correct?
if so, questions:

1- shouldn't this be done by sdk by default?
2- is this `coins.push` gonna be reflected in coins outputs? I'm asking because it has no `to` address
3- I see you're passing `minGasPrice`, which can be 1 unit for example. if UTXO input is of let's say 2 units, but real gasFee os transaction is 10 units, then this transaction will raise an error

Reply:
This coins are the resources needed to the transaction. It mean you need to account for the output coins but also the gasFee coins, when sending a alt asset, the outputs will not include ETH. Than the logic is if the coins required to the TX not include ETH, than we add ETH to the coins with the min amount to pay to the minGasFee. Than the SDK will check if this is enough to pay the gasUsed if not it automatically adds the amount required.

This logics we are doing to create a transfer can be removed and improve on the SDK but for now I tried to fix directly on the Wallet.